File: Assignment_14/program3.java
// Input 5 5

/*
    a       b       c       d       e
    1       2       3       4       5
    a       b       c       d       e
    1       2       3       4       5
    a       b       c       d       e
*/
import java.util.*;

class Pattern
{
    public void Display(int iRow, int iCol)
    {
        int i = 0, j = 0;
        char ch = '\0';

        for(i = 1; i <= iRow; i++)
        {
            for(j = 1, ch = 'a'; j <= iCol; j++, ch++)
            {
                if((i % 2) != 0)
                {
                    System.out.print(ch+"\t");
                }
                else
                {
                    System.out.print(j+"\t");
                }               
            }
            System.out.println();
        }
    }
}

class program1
{
    public static void main(String A[])
    {
        int iValue1 = 0, iValue2 = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter rows and columns: ");
        iValue1 = sobj.nextInt();
        iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();

        pobj.Display(iValue1,iValue2);
    }
}

//  Time Complexity = O(N)

File: Assignment_17/program3.c
// input : 6 6

/*
    *       *       *       *       *       *
    *                               *       *
    *                       *               *
    *               *                       *
    *       *                               *
    *       *       *       *       *       *
*/

#include<stdio.h>

void Pattern(int iRow, int iCol)
{
    int i = 0, j = 0;

    for(i = 1; i <= iCol; i++)
    {
        for(j = iRow; j >= 1; j--)
        {
            if((i == j) || (i == 1) || (i == iCol) || (j == 1) || (j == iRow))
            {
                printf("* \t");
            }
            else
            {
                printf(" \t");
            }

        }
        printf("\n");
    }

}

int main()
{
    int iValue1 = 0, iValue2 = 0;

    printf("Enter Rows and Columns : \n");
    scanf("%d %d",&iValue1,&iValue2);

    Pattern(iValue1, iValue2);

    return 0;
}